integration_tests/src/main/python/parquet_test.py

```diff
@@ -728,3 +728,39 @@ def do_parquet_scan(spark):
         exist_classes= "BatchScanExec",
         non_exist_classes= "GpuBatchScanExec",
         conf = conf_for_parquet_aggregate_pushdown)
+
+@pytest.mark.skipif(is_before_spark_330(), reason='Hidden file metadata columns are a new feature of Spark 330')
+@allow_non_gpu(any = True)
+def test_parquet_scan_with_hidden_metadata_fallback(spark_tmp_path):
+    data_path = spark_tmp_path + "/hidden_metadata.parquet"
+    with_cpu_session(lambda spark : spark.range(10) \
+                     .selectExpr("id", "id % 3 as p") \
+                     .write \
+                     .partitionBy("p") \
+                     .mode("overwrite") \
+                     .parquet(data_path))
+
+    def do_parquet_scan(spark):
+        df = spark.read.parquet(data_path).selectExpr("id", "_metadata.file_path", "_metadata.file_name",
+                                                      "_metadata.file_size", "_metadata.file_modification_time")
```

Comment:
It might be good to iterate the test through the possible metadata names, otherwise just recognizing a single metadata column name but missing the others would still allow this test to pass.